src/pages/home/report/ReportActionItemMessageEdit.js

```diff
@@ -96,16 +101,19 @@ class ReportActionItemMessageEdit extends React.Component {
     /**
      * Save the draft of the comment. This debounced so that we're not ceaselessly saving your edit. Saving the draft
      * allows one to navigate somewhere else and come back to the comment and still have it in edit mode.
+     * @param {String} newDraft
      */
-    debouncedSaveDraft() {
-        saveReportActionDraft(this.props.reportID, this.props.action.reportActionID, this.state.draft);
+    debouncedSaveDraft(newDraft) {
+        saveReportActionDraft(this.props.reportID, this.props.action.reportActionID, newDraft);
     }
 
     /**
      * Save the draft of the comment to be the new comment message. This will take the comment out of "edit mode" with
      * the new content.
      */
     publishDraft() {
+        this.debouncedSaveDraft.cancel();
```

Comment:
Please add a comment explaining why this cancel needs to be here.

Reply:
Hey @tgolen, I've added comments for both. Since English is not my native language, I may have expressed myself poorly. Please let me know if you want to reword anything.